core/shared/src/main/scala/zio/metrics/Metric.scala

```diff
@@ -436,23 +436,28 @@ object Metric {
         new UnsafeAPI {
           def update(in: key.keyType.In, extraTags: Set[MetricLabel] = Set.empty)(implicit
             unsafe: Unsafe
-          ): Unit =
-            hook(extraTags).update(in)
+          ): Unit = {
+            val fullKey = key.tagged(extraTags).asInstanceOf[MetricKey[key.keyType.type]]
```

Comment:
Computing the `fullKey` was not necessary before, and will involve additional allocation on every update of every metric.

Can you explain why this is necessary? At the very least, it should be fast-pathed, so that if `extraTags.isEmpty`, then no `fullKey` is computed.